rinad/src/ipcm/main.cc

```diff
@@ -63,6 +64,10 @@ void handler(int signum)
 		case SIGHUP:
 			rinad::IPCManager->stop();
 			break;
+		case SIGCHLD:
+			waitpid(WAIT_ANY, NULL, WNOHANG);
+			LOG_DBG("Child IPC Process Daemon died, removed from process table");
```

Comment:
I would get the result value, to check if it was successful (>0) or if no children was terminated (0), so that we can emit the correct LOG output. Also, I would call waitpid in a loop, to be more general, and exit the loop when waitpid returns 0 (no more children to reap).